fili-security/src/main/java/com/yahoo/bard/webservice/web/security/RoleDimensionApiFilterRequestMapper.java

```diff
@@ -0,0 +1,161 @@
+            "resource.  Please contact your user administration for access.";
+
+    private final static String UNAUTHORIZED_USER_MESSAGE = SYSTEM_CONFIG.getStringProperty(
+            SYSTEM_CONFIG.getPackageVariableName(SECURITY_SIGNUP_MESSAGE_KEY),
+            DEFAULT_SECURITY_MESSAGE
+    );
+
+    private final String unauthorizedHttpMessage;
+
+    Dimension dimension;
+    Map<String, Set<ApiFilter>> roleApiFilters;
+
+    /**
+     * Constructor.
+     *
+     * @param resourceDictionaries  The dictionaries to use for request mapping.
+     * @param dimension  The dimension whose roles are being matched
+     * @param roleApiFilters  ApiFilters by role for a given dimension
+     */
+    public RoleDimensionApiFilterRequestMapper(
+            ResourceDictionaries resourceDictionaries,
+            Dimension dimension,
+            Map<String, Set<ApiFilter>> roleApiFilters
+    ) {
+        this(resourceDictionaries, dimension, roleApiFilters, null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param resourceDictionaries  The dictionaries to use for request mapping.
+     * @param dimension  The dimension whose roles are being matched
+     * @param roleApiFilters  ApiFilters by role for a given dimension
+     * @param next  The next request mapper to process this ApiRequest
+     */
+    public RoleDimensionApiFilterRequestMapper(
+            final ResourceDictionaries resourceDictionaries,
+            Dimension dimension,
+            Map<String, Set<ApiFilter>> roleApiFilters,
+            ChainingRequestMapper<DataApiRequest> next
+    ) {
+        super(resourceDictionaries, next);
+        this.dimension = dimension;
+        this.roleApiFilters = roleApiFilters;
+        unauthorizedHttpMessage = DIMENSION_MISSING_MANDATORY_ROLE.format(dimension.getApiName());
+    }
+
+    @Override
+    public DataApiRequest internalApply(DataApiRequest request, ContainerRequestContext context)
+            throws RequestValidationException {
+        SecurityContext securityContext = context.getSecurityContext();
+
+        Set<ApiFilter> mergedSecurityFilters = unionMergeFilterValues(
+                roleApiFilters.keySet().stream()
+                        .filter(securityContext::isUserInRole)
+                        .map(roleApiFilters::get)
+                        .flatMap(Set::stream)
+        );
+
+        if (mergedSecurityFilters.size() == 0) {
+            String name = securityContext.getUserPrincipal().getName();
+            LOG.warn(DIMENSION_MISSING_MANDATORY_ROLE.logFormat(name, dimension.getApiName()));
+            throw new RequestValidationException(
+                    Response.Status.FORBIDDEN,
+                    unauthorizedHttpMessage,
+                    UNAUTHORIZED_USER_MESSAGE
+            );
+        }
+        Function<Map.Entry<Dimension, Set<ApiFilter>>, Set<ApiFilter>> substituteFilters = entry ->
```

Comment:
Much as I love me some in-line functions, we might be better off making this an instance method that takes the entry and `mergedSecurityFilters` as arguments. It'll save us having to build this function over and over again when building security filters. Normally I wouldn't care, but if customers are using it, this code will probably get called on _every_ request, and since I think both an instance method and an in-line function are equally simple and readable in this case...